deltacat/aws/lakeformation/lakeformation_table_reader.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import time
+import datetime
+import ray
+import pyarrow
+from typing import Optional, Callable, TypeVar
+from ray.data.impl.block_list import BlockList
+from ray.data.block import BlockMetadata
+from deltacat.aws.lakeformation.lakeformation_utils import ( 
+    get_lakeformation_client,
+    merge_blocklists
+)
+
+T = TypeVar('T')
+
+logger = logging.getLogger(__name__)
+
+@ray.remote
+def get_work_unit_result(region: str,
+                        query_id: str, work_unit_id: int, work_unit_token: str,
+                        mapper: Callable[[pyarrow.Table, str, str, dict], T],
+                        role_arn: str, **kwargs) -> T:
+
+    client = get_lakeformation_client(region, role_arn)
```

Comment:
Building a client everytime you get the work unit result will be expensive.

Reply:
This is the only way we have as of now. Passing a client results in serialisation error.  check - https://discuss.ray.io/t/raysgd-pytorch-fail-typeerror-cant-pickle-sslcontext-objects/3921